### ISSUE TITLE
[java backend] Multiple catch block bug

### ISSUE THREAD

**Issue by exFalso** [HaxeFoundation/haxe#4134]
If there are multiple catch blocks and the caught exceptions have an inheritance relation the catch block to be executed is wrong. See the followin gist: https://gist.github.com/exFalso/a7e3bfbcce2c2b53cd49

Just to repeat the relevant code:
```
		try
		{
			throw new B();
		}
		catch(b : B)
		{
			trace("first");
		}
		catch(a : A)
		{
			trace("second");
		}
```
where ```B``` inherits from ```A```
For js it prints "first", for java "second".

**Comment by Simn** [comment]
With any luck this is just a reversed list order issue.

**Comment by Simn** [comment]
I have pushed a fix but I'll need @waneck to review it for 3.2.

**Comment by waneck** [comment]
That was it. Thanks!!
